src/lib/node.ts

```diff
@@ -209,3 +209,75 @@ export function pipeToNodeStream<T>(
 		handleTsStreamError // abort handler
 	);
 }
+
+/**
+ * Create a ts-stream ReadableStream from a Node.JS ReadableStream.
+ *
+ * Reads all values from `nodeReadable` and writes them to the returned stream.
+ * When the `nodeReadable` ends, the returned stream is also ended.
+ * Should the `nodeReadable` emit a `close` event, the returned stream is ended without error.
+ * If the `nodeReadable` emits a `error` event, the returned stream is ended with the given error.
+ *
+ * Usage example:
+ * ```ts
+ * const source = fromNodeReadable(fs.createReadStream('source.txt'));
+ * source
+ *  .map((chunk) => String(chunk))
+ *  .map((letters) => letters.toUpperCase())
+ *  .forEach(console.log);
+ * ```
+ *
+ * @param nodeReadable Soource NodeJS stream
+ * @returns Readable stream of type `string | Buffer`
+ */
+export function fromNodeReadable(
+	nodeReadable: NodeJS.ReadableStream
+): ReadableStream<string | Buffer> {
```

Comment:
Great callout! I've updated the type signature to support an optional generic type argument.

```typescript
// creates a ts-stream of type `Stream<MyObjectType>`
const withObjectMode = fromNodeReadable<MyObjectType>(someStreamInObjectMode);

// creates a ts-stream of type `Stream<string | Buffer>` since the caller did not specify a type
const noObjectmode = fromNodeReadable(someStreamNotInObjectMode);
```

Follow-up questions: do you think this is a potential foot gun? Should I include some documentation around this? I could imagine that someone might be working with a stream in object mode but fail to specify the type for `fromNodeReadable` which means the ts-stream will work, but the type may be incorrect.

Reply:
> do you think this is a potential foot gun

In a way, yes, but I think it's the best way to do it given that the Node typings don't provide any way of correctly typing object mode streams...

Including the lines you show here in the docs is a very good idea indeed!